File: modules/wem-webapp/src/main/webapp/admin/common/js/content/page/PageTemplateKey.ts
module api.content.page {

    export class PageTemplateKey {

        public static SEPARATOR: string = "|";

        private moduleName: string;

        private templateName: PageTemplateName;

        private refString: string;

        public static fromString(str: string): PageTemplateKey {

            var elements: string[] = str.split(api.content.page.PageTemplateKey.SEPARATOR);
            var moduleName = elements[0];
            var templateName = new PageTemplateName(elements[1]);
            return new PageTemplateKey(moduleName, templateName);
        }

        constructor(moduleName: string, templateName: PageTemplateName) {
            if (name == null) {
                throw new Error("name cannot be null");
            }
            this.moduleName = moduleName;
            this.templateName = templateName;
            this.refString = this.moduleName + PageTemplateKey.SEPARATOR + this.templateName;
        }

        getModuleName(): string {
            return this.moduleName;
        }

        getTemplateName(): PageTemplateName {
            return this.templateName;
        }

        toString(): string {
            return this.refString;
        }

        equals(other: PageTemplateKey) {
            return other && (this.refString === other.refString);
        }
    }
}

File: modules/core/core-api/src/main/java/com/enonic/xp/page/PageTemplateKey.java
package com.enonic.xp.page;

import com.enonic.xp.content.ContentId;


public final class PageTemplateKey
{
    private final ContentId id;

    private PageTemplateKey( final ContentId id )
    {
        this.id = id;
    }

    public ContentId getContentId()
    {
        return this.id;
    }

    @Override
    public boolean equals( final Object o )
    {
        return this == o || o instanceof PageTemplateKey && this.id.equals( ( (PageTemplateKey) o ).id );
    }

    @Override
    public int hashCode()
    {
        return this.id.hashCode();
    }

    @Override
    public String toString()
    {
        return this.id.toString();
    }

    public static PageTemplateKey from( final String value )
    {
        return from( ContentId.from( value ) );
    }

    public static PageTemplateKey from( final ContentId value )
    {
        return new PageTemplateKey( value );
    }
}
